lib/AccessSystem/API/Controller/Root.pm

```diff
@@ -228,6 +235,33 @@ sub delete_me :Chained('logged_in'): PathPart('deleteme'): Args(0) {
     }
 }
 
+sub delete_token :Chained('logged_in'): PathPart('delete_token'): Args(0) {
+    my ($self, $c) = @_;
+    my $token = $c->req->params->{token};
+
+    # These are GET reqs (cos lazy, and its a link)
+    # Must retain at least one token
+    if($c->stash->{person} && $c->stash->{person}->tokens_rs->count > 1) {
```

Comment:
This doesn't *retain* one token.  On the other hand, I'm not sure why we want to force users to retain one token anyway?

Reply:
It does.. if you try and delete, and you only have 1, it doesn't allow it